Copilot for finance custom connector

elkuliko opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

elkuliko committed:
adding copilot for finance custom connector

elkuliko committed:
nit changes

elkuliko committed:
nit changes

elkuliko committed:
add identity provider option

elkuliko committed:
remove space

vmanoharas commented:
Hello @elkuliko, 

Could you please go through the readme file from the below url and prepare files? 

https://github.com/microsoft/PowerPlatformConnectors, files you have submitted are incorrect, kindly resubmit.

vmanoharas commented:
> Hello @elkuliko,
> 
> Could you please go through the readme file from the below url and prepare files?
> 
> https://github.com/microsoft/PowerPlatformConnectors, files you have submitted are incorrect, kindly resubmit.

Hello @elkuliko, 

I am following up on my above comments, kindly resubmit your files by following the instructions provided.

elkuliko committed:
add readme file

elkuliko commented:
> Hello @elkuliko,
> 
> Could you please go through the readme file from the below url and prepare files?
> 
> https://github.com/microsoft/PowerPlatformConnectors, files you have submitted are incorrect, kindly resubmit.

Was the issue missing Readme file? I've added the Readme.
Renamed swagger file as well.
@vmanoharas  Please let me know if something else is required.

elkuliko committed:
rename swagger

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/Readme.md:
```diff
@@ -0,0 +1,9 @@
+
```
Please format readme.md file as per the below link, and ensure that all required sections are updated. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md

elkuliko replied:
Updated

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiProperties.json:
```diff
@@ -0,0 +1,16 @@
+{
```
Please refer the below schema link and update this file. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/paconn-apiProperties.schema.json

elkuliko replied:
updated

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiProperties.json:
```diff
@@ -0,0 +1,16 @@
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
```
please make sure you are using the iconBrandColor for your connector. Icon color must contain a non-transparent, non-white color (#ffffff) background and not-default color (#007ee5) that matches your specified icon background color. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector).

vmanoharas reviewed (commented):
Hello @elkuliko, 

Kindly resolve review comments and can you make sure that this connector is the update for "Azure Communication Chat" which is already exist?

elkuliko committed:
fix comments

elkuliko commented:
> Hello @elkuliko,
> 
> Kindly resolve review comments and can you make sure that this connector is the update for "Azure Communication Chat" which is already exist?

Sorry. I've accidentally included incorrect swagger file in the previous iteration. This connector is a new one, not related to "Azure communications"

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-    "x-ms-api-annotation": {
-      "status": "Production"
-    },
-    "contact": {
```
Contact is mandatory, kindly add it again.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-        },
-        "summary": "List chat thread messages",
-        "description": "List messages From chat thread.",
-        "operationId": "ListMessages",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/ListMessages

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-        },
-        "summary": "Send message to chat thread",
-        "description": "Send message to chat thread.",
-        "operationId": "SendChat",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/SendChat

elkuliko replied:
We are adding a new template for a custom connector, so nothing is actually deleted.
Endpoint is deleted compared to prior iteration because prior iteration had incorrect swagger included.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-        },
-        "summary": "Add chat thread participants",
-        "description": "Add participants to chat thread.",
-        "operationId": "AddParticipants",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/AddParticipants

elkuliko replied:
We are adding a new template for a custom connector, so nothing is actually deleted.
Endpoint is deleted compared to prior iteration because prior iteration had incorrect swagger included.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-        },
-        "summary": "Remove chat thread participant",
-        "description": "Remove participant from chat thread.",
-        "operationId": "RemoveParticipant",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/RemoveParticipant

elkuliko replied:
We are adding a new template for a custom connector, so nothing is actually deleted.
Endpoint is deleted compared to prior iteration because prior iteration had incorrect swagger included.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-        },
-        "summary": "List chat threads",
-        "description": "List chat threads.",
-        "operationId": "ListChatThreads",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/ListChatThreads

elkuliko replied:
We are adding a new template for a custom connector, so nothing is actually deleted.
Endpoint is deleted compared to prior iteration because prior iteration had incorrect swagger included.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-        },
-        "summary": "Create chat thread",
-        "description": "Create a chat thread.",
-        "operationId": "CreateChat",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/CreateChat

elkuliko replied:
We are adding a new template for a custom connector, so nothing is actually deleted.
Endpoint is deleted compared to prior iteration because prior iteration had incorrect swagger included.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
-        },
-        "summary": "List chat thread participants",
-        "description": "List participants in chat thread.",
-        "operationId": "ListParticipants",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/ListParticipants

elkuliko replied:
We are adding a new template for a custom connector, so nothing is actually deleted.
Endpoint is deleted compared to prior iteration because prior iteration had incorrect swagger included.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiDefinition.swagger.json:
```diff
@@ -1,765 +1,1376 @@
+            }
           }
-        },
-        "operationId": "GetThreadProperties",
```
The operation has been deleted. This change breaks existing flows. ValuePath: operation/GetThreadProperties

elkuliko replied:
We are adding a new template for a custom connector, so nothing is actually deleted.
Endpoint is deleted compared to prior iteration because prior iteration had incorrect swagger included.

vmanoharas commented on custom-connectors/CopilotForFinanceCommunications/apiProperties.json:
```diff
@@ -2,13 +2,18 @@
     "connectionParameters": {
       "token": {
-        "type": "oauthSetting",
+        "type": "oAuthSetting",
```
Error; message=Value "oAuthSetting" is not defined in enum.; valuePath=line: 5 position: 30 path: '[0].properties.connectionParameters.token.type'; value=oAuthSetting

vmanoharas replied:
Thank you very much for resolving review comments, kindly resolve this comment as well. @elkuliko

vmanoharas reviewed (commented):
Hello @elkuliko, 

Kindly fix review comments additionally, please refer the below schemas while resolving the comments. 

https://github.com/microsoft/PowerPlatformConnectors/tree/dev/schemas

elkuliko committed:
fix APIProperties

vmanoharas reviewed (approved):
Dear Partner, 

Congratulations your PR is approved !! We are proud to announce the Brand-New Certification experience for you to certify Power Platform Copilot Connectors & Plugins. Read the blog here- [Announcing Partner Center to certify and publish Power Platform Copilot Connectors and Plugins - Microsoft Power Platform Blog](https://www.microsoft.com/en-us/power-platform/blog/power-automate/announcing-partner-center-to-certify-and-publish-power-platform-copilot-connectors-and-plugins/)

As next steps: 

1.	Learn about the new certification experience- [Get your Power Platform connector and plugin certified - Overview](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390456966%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=uWDvs%2BmTbozw1oPc%2FNDNWJ%2FHgnUSfFlRCB1T8UHBh3A%3D&reserved=0)
2.	Package your connector and plugin files- [Prepare Power Platform connector and plugin files for certification](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-submission&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390465418%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=pDuv4bnya1eP3Rrs3p4ArNOXw%2FQipWU87q5oZ0VSIe0%3D&reserved=0)
3.	Initiate a certification request in Partner Center- [Verified publisher certification process](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-for-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390471799%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=zbm2px5VZ2ABLBFcORH6tu1fsC9KWdXkP1XJFyluMzw%3D&reserved=0)
4.	Ensure your connector & plugin files comply to the Marketplace policies here- [5000 Power Platform Connector Policies for Marketplace](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Flegal%2Fmarketplace%2Fcertification-policies%235000-power-platform-connectors&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390477736%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=h0pI5J5buo82xxvnBVmrAN8K%2F%2BK8xZlUdcSwzWoqWq0%3D&reserved=0) and also [1000 Marketplace policies](https://learn.microsoft.com/en-us/legal/marketplace/certification-policies#100-general)(as applicable) 
5.	Test your connector post certification to provide go-live for deployment- [Testing Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-testing&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390483743%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=Z8pSnhCArFRzEr373o29qM5h5VbV1nlN2gVbv%2Blk1aE%3D&reserved=0)
6.	If you wish to Update your connector or plugin in future- [Updating Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-updates&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390489681%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=CvxTWJFE3bTFeJ3nDJoCSpzvimhVV9go4cBG%2FQ4gMvo%3D&reserved=0)

For any queries or concerns please get in touch with us on [certificationteam@microsoft.com](mailto:certificationteam@microsoft.com)

Many thanks,
Power Platform 
Copilot connector & plugin certification team
